ols/user_data_collection/data_collector.py

```diff
@@ -0,0 +1,323 @@
+logging.getLogger("kubernetes").setLevel(logging.WARNING)
+logging.getLogger("urllib3").setLevel(logging.WARNING)
+
+logger = logging.getLogger(__name__)
+
+
+class ClusterPullSecretNotFoundError(Exception):
+    """Cluster pull-secret is not found."""
+
+
+def get_ingress_upload_url() -> str:
+    """Get the Ingress upload URL based on the environment."""
+    upload_endpoint = "api/ingress/v1/upload"
+    if INGRESS_ENV == "prod":
+        return "https://console.redhat.com/" + upload_endpoint
+    elif INGRESS_ENV == "stage":
+        return "https://console.stage.redhat.com/" + upload_endpoint
+    else:
+        raise ValueError(f"Unknown value in INGRESS_ENV: {INGRESS_ENV}")
+
+
+def access_token_from_offline_token(offline_token: str) -> str:
+    """Generate "access token" from the "offline token".
+
+    Offline token can be generated at:
+        prod - https://access.redhat.com/management/api
+        stage - https://access.stage.redhat.com/management/api
+
+    Args:
+        offline_token: Offline token from the Customer Portal.
+
+    Returns:
+        Refresh token.
+    """
+    if INGRESS_ENV == "stage":
+        url = "https://sso.stage.redhat.com/"
+    else:
+        url = "https://sso.redhat.com/"
+    endpoint = "auth/realms/redhat-external/protocol/openid-connect/token"
+    data = {
+        "grant_type": "refresh_token",
+        "client_id": "rhsm-api",
+        "refresh_token": offline_token,
+    }
+
+    response = requests.post(url + endpoint, data=data, timeout=REDHAT_SSO_TIMEOUT)
+    if response.status_code == requests.codes.ok:
+        return response.json()["access_token"]
+    else:
+        raise Exception(f"Failed to generate access token. Response: {response.json()}")
+
+
+def get_cloud_openshift_pull_secret() -> str:
+    """Get the pull secret token from the cluster."""
+    kubernetes.config.load_incluster_config()
+    v1 = kubernetes.client.CoreV1Api()
+
+    # get the pull-secret from openshift-config namespace
+    secret = v1.read_namespaced_secret("pull-secret", "openshift-config")
+    try:
+        dockerconfigjson = secret.data[".dockerconfigjson"]
+        dockerconfig = json.loads(base64.b64decode(dockerconfigjson).decode("utf-8"))
+        return dockerconfig["auths"]["cloud.openshift.com"]["auth"]
+    except KeyError:
+        logger.error("failed to get token from cluster pull-secret")
+        raise ClusterPullSecretNotFoundError
+
+
+def get_cluster_id() -> str:
+    """Get the cluster_id from the cluster."""
+    kubernetes.config.load_incluster_config()
+    custom_objects_api = kubernetes.client.CustomObjectsApi()
+
+    # get the cluster version object
+    version_data = custom_objects_api.get_cluster_custom_object(
+        "config.openshift.io", "v1", "clusterversions", "version"
+    )
+    try:
+        return version_data["spec"]["clusterID"]
+    except KeyError:
+        logger.error("failed to get cluster_id from cluster")
+        return "unknown"
```

Comment:
for the pullsecret helper you raise an exception, but for this helper you return "unknown", i think you should be consistent, and i think raising the exception is the right thing to do.  Checking for magic string values is fragile and puts an unreasonable burden on the caller to anticipate that return value.